worldengine/draw.py

```diff
@@ -168,43 +168,33 @@ def draw_simple_elevation(world, sea_level, target):
     """ This function can be used on a generic canvas (either an image to save
         on disk or a canvas part of a GUI)
     """
-    min_elev_sea = None
-    max_elev_sea = None
-    min_elev_land = None
-    max_elev_land = None
-    for y in range(world.height):
-        for x in range(world.width):
-            e = world.elevation['data'][y,x]
-            if sea_level is None:
-                if min_elev_land is None or e < min_elev_land:
-                    min_elev_land = e
-                if max_elev_land is None or e > max_elev_land:
-                    max_elev_land = e
-            elif world.is_land((x, y)):
-                if min_elev_land is None or e < min_elev_land:
-                    min_elev_land = e
-                if max_elev_land is None or e > max_elev_land:
-                    max_elev_land = e
-            else:
-                if min_elev_sea is None or e < min_elev_sea:
-                    min_elev_sea = e
-                if max_elev_sea is None or e > max_elev_sea:
-                    max_elev_sea = e
+    e = world.elevation['data']
+
+    c = numpy.empty(e.shape, dtype=numpy.float)
+    if sea_level is None or world.ocean is None or not world.ocean.any():  # or 'not any ocean'
+        mask = numpy.ma.array(e, mask = False)  # the whole map
+        min_elev_land = mask.min()
+        max_elev_land = mask.max()
+        elev_delta_land = (max_elev_land - min_elev_land) / 11.0
 
-    elev_delta_land = (max_elev_land - min_elev_land)/11
-    if sea_level != None:
+        c = ((e - min_elev_land) / elev_delta_land) + 1
+    else:
+        mask = numpy.ma.array(e, mask = world.ocean)  # only land
+        min_elev_land = mask.min()
+        max_elev_land = mask.max()
+        elev_delta_land = (max_elev_land - min_elev_land) / 11.0
+
+        mask = numpy.ma.array(e, mask = numpy.logical_not(world.ocean))  # only ocean
```

Comment:
perhaps it could be clearer to use mask_land and mask_ocean instead of reusing mask